### PR TITLE
Support minimal mode.

### PR DESCRIPTION
**Pull request by yangyingchao**
When using eglot in Emacs, I noticed that `nc` is slower than `cmake-language-server`, which is implemented with JavaScript.

After conducting some investigation, it appears that `nc` returns a significantly larger amount of data compared to `cmake-language-server`.

Therefore, I made some modifications to `nc` in order to reduce the amount of data it returns to the client. This was achieved by implementing server-side filters and decreasing the level of detail in completion items.

This functionality can be activated by running `neocmakelsp --stdio --minimal`.

I am uncertain about whether this pull request will benefit others. Please do not hesitate to close it if you feel that this modification deviates from your intended design.

### CODE REVIEW

File: src/complete/buildin.rs [hunk 1]
```diff
@@ -1,10 +1,37 @@
 /// buildin Commands and vars
 use anyhow::Result;
 use once_cell::sync::Lazy;
+use std::env;
 use std::process::Command;
 use std::{collections::HashMap, iter::zip};
 use tower_lsp::lsp_types::{CompletionItem, CompletionItemKind};
 
+pub static CMAKE_MINIMAL: Lazy<bool> = Lazy::new(|| {
+    let key = "NEOCMAKELSP_MINIMAL";
```
Comment (Decodetalkers) [inline src/complete/buildin.rs:10]: you should use https://github.com/Decodetalkers/neocmakelsp/blob/master/src/languageserver/config.rs lsp config to control it, not environment, I think..

**Comment by Decodetalkers** [comment]
I think the detail part should be the same as cmake-language-server.. but it is faster when using emacs.. Can you provide me a video? I want to confirm it

**Comment by Decodetalkers** [comment]
and it is better to use lsp init_options to control it. I think

**Comment by Decodetalkers** [comment]
Emm why close it? Emm. I will feel glad to check this problem. As I do not use emacs, so I do not know what happened on that editor

**Comment by yangyingchao** [comment]
I'm going to rework on it when I have time, and then reopen it :)

**Pull request closed without merging**